students/csimmons/lesson03/mailroom.py

```diff
@@ -77,29 +79,25 @@ def generate_thankyou(f_response, f_gift):
 def send_thankyou(donorlist):
     donors = list(map(lambda x:x[0], donorlist))
     response = input(thanks_prompt)
-    f_response = response.title()
     if response.lower() == 'list':
         print_donors(donors)
-        main()
-    elif response.lower() == 'exit':
-        main()
+    elif response.lower() == 'exit': 
+        menu()
     elif response.title() in donors:
         exist_donor(response, donors)
-        main()
     else:
         new_donor(response)
-        main()
 
 def display_report(donorlist):
     donors = list(map(lambda x:x[0], donorlist))
     gifts = list(map(lambda x:x[1], donorlist))
     all_info = []
-    for i in range(len(donorlist)):
+    for i, donor in enumerate(donorlist):
```

Comment:
range() or enumerate() are unnecessary for your use case here.

```
for name, donations in donorlist:
    total = sum(donations)
    num_gifts = len(donations)
    avg = total / num_gifts
    gift_info.append((total, num_gifts, avg))
```

notice how simple and readable code is ^^